### DISCUSSION TITLE
How to connect the authoring views

### DISCUSSION THREAD

**Discussion by mwpowellhtx** [OrchardCMS/OrchardCore#11220] [Q&A] [answered]
Using the `Coming Soon` boilerplate stuff initially. I see what I think are the rendered outputs. Is there a way to review the _admin_ or _author_ level views, however? I see I can access an `https://localhost:5001/admin` page, for instance, but I do not see anything that indicates the _content_, _items_, _views_, or other half-expected CMS artifacts, etc. Apologies if my verbiage is a bit clumsy. Thanks!

**Comment by Piedone** [comment]
Not really sure what your question is about, but are you asking where the display (HTML and else) of admin pages are coming from? They are from [TheAdmin](https://github.com/OrchardCMS/OrchardCore/tree/main/src/OrchardCore.Themes/TheAdmin) theme and from the respective modules. E.g. [this template](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Title/Views/TitlePart.Edit.cshtml) is for the title editor.

**Comment by mwpowellhtx** [comment]
I see this message in green: _Feel free to browse the menu and discover all its possibilities._

However, I do not see any menus in which to edit anything. Perhaps I am doing something incorrectly.

Edit: Indeed, reviewing the [code behind that](https://github.com/OrchardCMS/OrchardCore/blob/5310bdb49797e9db02f3107ea9c0feb270e92346/src/OrchardCore.Modules/OrchardCore.Admin/Views/AdminDashboard.cshtml), there does not appear to be any mention of any menus, like legit they are not there. Was this a boilerplate placeholder, perhaps?

Are we better off starting from a green field workflow and rolling our own `'coming soon'` version, perhaps?

**Reply by Piedone** [reply]
Can you add a screenshot? To the left of that you should see the admin menu.
![image](https://user-images.githubusercontent.com/1976647/154745346-4a6c64d0-15c2-4214-ac3b-a3b172256cfe.png)

**Reply by mwpowellhtx** [reply]
> Can you add a screenshot? To the left of that you should see the admin menu. ![image](https://user-images.githubusercontent.com/1976647/154745346-4a6c64d0-15c2-4214-ac3b-a3b172256cfe.png)

I do not have that menu on the left. Again, starting from the `Coming Soon` boilerplate.

**Reply by Piedone** [reply]
The Coming Soon adds that menu as well, just a little shorter. Could you share a full-window screenshot? As well as is there anything under App_Data\logs?

**Reply by mwpowellhtx** [reply]
This is 'plain Jane' out of the box. There are no logs. This is exactly what I am seeing, as I said.
![image](https://user-images.githubusercontent.com/1824187/154748282-944c447a-ed89-40d0-8b27-71992d35ad2c.png)

**Reply by mwpowellhtx** [reply]
I might also note, the _company name_ is not reflected anywhere, either. `MvcSandbox` versus the name I entered on the input side.

**Comment by hishamco** [comment]
> I see this message in green: Feel free to browse the menu and discover all its possibilities.

The message appears in the dashboard, if you don't see the menu please attach a screenshot I may have a look, sometimes it happens if there's a js bug or a caching issue, may be there's something wrong there

**Comment by mwpowellhtx** [comment]
Plus side `admin` did at least prompt me for credentials, so that much is 'working'.

**Reply by hishamco** [reply]
I will have a look, please let me know if you are using the source or a specific NuGet package to make things clear to me

**Reply by mwpowellhtx** [reply]
Using the 'latest' `1.2.2` NuGet production release package, AFAIK. Thank you.

**Reply by hishamco** [reply]
I need to add some commits in one of my PR, then I will have a look

**Reply by mwpowellhtx** [reply]
I get this warning in the console:

```
DevTools failed to load source map: Could not load content for https://ajax.aspnetcdn.com/ajax/jquery/jquery.min.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
```

And it does not load anything else, perhaps?

**Reply by hishamco** [reply]
As I told you, so please wait until I investigate on that

**Reply by mwpowellhtx** [reply]
Another data point for you perhaps. No worries. I am paused in the workflow evaluation. Thank you.

**Comment by mwpowellhtx** [comment]
~In summary, as I mentioned, evaluating whether `OrchardCore/CMS` may be ready to adopt for our purposes... In terms of not only the technology stacks, which _'I know'_ so to say, but more importantly, _'is it ready to start being productive generating content from a business perspective'_, let alone any other language level business enhancements.~

~Any ideas why none of the input details are connected with the rendered pages? I gather there is 'something' that needs to be connected there, somehow. Out of the box, all I can say is, it is a non-starter at the moment, unless/until those dots have been connected.~
~1. Ability to admin the CMS itself~
~1. Connecting the input details with the rendered pages themselves~
~1. Ability to identify the `Site` template as something (anything) else other than `'Default'`, i.e. . But maybe that is the admin menu wiring that exposes that, I do not know, cannot say.~

![22-02-19-orchardcore-cms-coming-soon-oob-rendering](https://user-images.githubusercontent.com/1824187/154809027-b867bfaa-00c8-4748-ab9b-5d689c6c2387.png)

~Using `postgres` for our database backend. Again, this is out of the box from the `1.2.2` NuGet package; I did not do anything special or other than following the template getting started, `Coming Soon`.~

~At the moment at least in the `Coming Soon` template, gauging readiness for adoption, scale of _1-10_, somewhere between a _3-5_; letter grade, something like a _C-_ possibly a _D+_. Now, if there is a 'better' template to start from that is more capable that empowers us to proceed, working around the issues I've organically stumbled upon, that's a different story.~

~I have also noticed in the _Chrome console warnings_, failing to load the _jquery minified assets_, for starters. Perhaps there are front end elements that are loaded in dynamically in that sense; but without the `jquery`, the dots fail to connect.~

```
Failed to load resource: the server responded with a status of 404 () - re: favicon.ico:1
DevTools failed to load source map: Could not load content for https://ajax.aspnetcdn.com/ajax/jquery/jquery.min.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
```

~I do not care about a _favicon_ at the moment. However the jquery issue seems like a snafu to me. The asset may well not exist via `https://ajax.aspnetcdn.com` sources? I do not know; again, from an adoption perspective, these are far deeper dives than we are willing to assume.~

*Edit*: Happily _'striking'_ these from the discussion. As mentioned below, turns out I was basing this mistakenly on an incorrect project template... So when the foundation isn't great, of course, this was the issue. Created a project instead based on the CMS project template, and that made all the difference. I just did not want to leave the impression that I was somehow making it work not working, if you will. Again, appreciate all the hints, it was certainly helpful.

**Reply by mwpowellhtx** [reply]
Following up with `Setup` side concerns. Hopefully that makes it a bit clearer.

![image](https://user-images.githubusercontent.com/1824187/154860192-1be2fab8-17ee-4ccc-9238-ddd9da324e32.png)

Also, the issue of 'scalability' for a smaller number of pages it is probably fine. But how does it scale into the hundreds or thousands of pages? At least as discovered by one (now defunct?) competitor, _Carnotaurus CMS_. In my experience, when we talk about a document database infrastructure, usually has to do with the kind of loading lazy/eager-ness, how well factored the document tables are, talking about _MartenDB_ for instance and some of its features, inclusion, filtering, etc. I do not know if it would be any better/worse for relational analogs, also depending on the normalization and similar loading policies.

Maybe it is better to start from a _blank slate_ recipe, perhaps; which would that be? And then build the sites, versions, etc, out from there. Have not tried that yet; open to suggestions, constructive feedback, etc.

**Reply by hishamco** [reply]
Ah, seems You are using `OrchardCore.Application.Mvc.Targets`? not `OrchardCore.Application.Cms.Targets`

**Comment by hishamco** [comment]
Nothing wrong with TheComingSoon theme, I just added a new tenant using this theme

![Screenshot 2022-02-20 222316](https://user-images.githubusercontent.com/3237266/154860470-acca85df-827a-4ae1-9c09-b261a5083da5.png)

![Screenshot 2022-02-20 222406](https://user-images.githubusercontent.com/3237266/154860509-36e1036a-0879-4a80-a638-f5d4465a981f.png)

**Reply by mwpowellhtx** [reply]
Using what as a backend? We are using `postgres`. Reported precisely what we got. It does not work.

**Reply by hishamco** [reply]
BTW I'm using the source, could you please send a gif showing tenant creating with TheComingSoon theme?

**Reply by mwpowellhtx** [reply]
> BTW I'm using the source...

Ah okay; yeah, we are _"not"_, we are running against released packages `1.2.2`.

> ...could you please send a gif showing tenant creating with TheComingSoon theme?

No. Not sure I follow there; what do you mean "TheComingSoon theme".

As I stated, we're not interested to cut our fingers on beta or preview quality code. We are interested in being productive to the business on released modules, `1.2.2` at the time of this response.

If there is an _RC_ or it is ready to push in a `1.2.3` or `1.3.0`, I'd encourage that trajectory instead.

**Reply by hishamco** [reply]
Postgres is just a database

**Reply by hishamco** [reply]
Running `OrchardCore.Application.Mvc.Targets` shows the `MvcSandbox` which is similar to what attached [here](https://github.com/OrchardCMS/OrchardCore/discussions/11220#discussioncomment-2210961)

**Reply by hishamco** [reply]
Please share the content of your `.csproj` content seems there's something wrong on the target

**Reply by mwpowellhtx** [reply]
In context here, got 'er sorted, I think. At least now _themes_ showing up as expected, `admin` includes the appropriate _menuing_, etc.

**Comment by ns8482e** [comment]
It seems you are not using CMS package.
Use this package for CMS https://www.nuget.org/packages/OrchardCore.Application.Cms.Targets/1.2.2

**Reply by hishamco** [reply]
That's what I meant in my previous thread

**Reply by mwpowellhtx** [reply]
Ah okay, could be? I will try that instead. Thank you.

These are my dependencies. I am halfway certain I started with the CMS targets.

![image](https://user-images.githubusercontent.com/1824187/154867110-febe00b7-5e8a-4f28-9689-d6f33ce75af1.png)

**Reply by mwpowellhtx** [reply]
I started from the `Orchard` project template, then added in the _CMS targets_. Should I just drop the _MVC targets_ then and replace with the _CMS_ ones?

**Reply by mwpowellhtx** [reply]
Okay, so that does not 'work' either, in fact, I get an exception dropping the _MVC targets_ package dependency. Apparently that is required.

```
An unhandled exception occurred while processing the request.
InvalidOperationException: The view 'Index' was not found. The following locations were searched:
/Areas/OrchardCore.Setup/Views/Setup/Index.cshtml
/Areas/OrchardCore.Setup/Views/Shared/Index.cshtml
/Views/Shared/Index.cshtml
/Pages/Shared/Index.cshtml
Microsoft.AspNetCore.Mvc.ViewEngines.ViewEngineResult.EnsureSuccessful(IEnumerable<string> originalLocations)
```

To recap, starting from the _OrchardCore project template_, added in the _CMS targets_.

**Reply by mwpowellhtx** [reply]
:+1: Now we're cooking with gas! Started from the _CMS_ and not the _MVC project template_, that was the difference.

Apologies for the confusion.

**Reply by hishamco** [reply]
All the best, hope it works without an issues ;)

**Reply by mwpowellhtx** [reply]
It's one step closer at any rate; appreciate the orientation.

**Reply by hishamco** [reply]
If you faced another related issue please let us know, otherwise make the right answer

**Comment by mwpowellhtx** [comment, marked as answer]
After some helpful hints, troubleshooting and [zeroing in on the issue](https://github.com/OrchardCMS/OrchardCore/discussions/11220#discussioncomment-2216925), turns out I had to start from the _CMS template_, not the _MVC template_. That sorted us out.

**Reply by hishamco** [reply]
Seems me then Niraj already said that and your mark your as answer ;)

**Reply by mwpowellhtx** [reply]
Starting from the project template? No, I think it was a reference to the NuGet packages, not the VS project templates. But the clues helped isolate the root cause, certainly.
